Program.cs

```diff
@@ -1,7 +1,120 @@
-﻿namespace Training {
-    internal class Program {
-        static void Main (string[] args) {
-            Console.WriteLine ("Hello, World!");
-        }
-    }
+﻿// ---------------------------------------------------------------------------------------
+// Training ~ A training program for new interns at Metamation, Batch - July 2023
+// Copyright (c) Metamation India.                                              
+// ---------------------------------------------------------------------------------------
+// Program.cs
+// Solve the famous 8 Queens problem
+// On a standard chessboard 8 queens have to be placed so that no queen can attack any other.
+// As per the rules of chess, the queen can move horizontally, vertically or in a diagonal.
+// No two queens can be on the same row or column or diagonal.
+// Find all the possible solutions. Each solution should display the Queens in 8x8 grid.
+// Extend the assignment to eliminate identical solutions. A solution is identical to another if it is:
+// 90, 180 or 270 degree rotation of the other
+// mirror image of the other : horizontal mirror, vertical mirror
+// ---------------------------------------------------------------------------------------
+using System.Text;
+using static System.Console;
+
+namespace Training {
+   #region Program --------------------------------------------------------------------------------
+   /// <summary>Eight Queens problem</summary>
+   class Program {
+      #region Method ------------------------------------------------
+      /// <summary>Print all the possible solutions or unique solutions</summary>
+      static void Main () {
+         Write ("Press U for unique solutions.");
+         bool isUni = false;
+         if (ReadKey (true).Key is ConsoleKey.U) isUni = true;
+         PlaceQueen (isUni, 0);
+         PrintChessBoard ();
+      }
+
+      /// <summary>Place Queens in safe cell</summary>
+      /// <param name="isUnique"></param>
+      /// <param name="row"></param>
+      static void PlaceQueen (bool isUnique, int row) {
+         for (int column = 0; column < 8; column++) {
+            if (IsSafeCell (row, column)) {
+               sPosition[row] = column;
+               var sln = sPosition.ToArray ();
+               if (isUnique ? row == 7 && IsUniqueSln (sln) : row == 7) sSolutionsList.Add (sln);
+               else PlaceQueen (isUnique, row + 1);
+            }
+         }
+
+         /// <summary>Checks if the cell is safe</summary>
+         static bool IsSafeCell (int row, int column) {
+            for (int preRow = 0; preRow < row; preRow++) {
+               int preColumn = sPosition[preRow];
+               if (Math.Abs (column - preColumn) == Math.Abs (row - preRow) || column == preColumn) return false;

```

Comment:
Add appropriate comments. "Check if the cell is safe" does not explain what this method checks. What is a safe cell? 
I feel most of the other methods are not commented properly. For instance, consider IsRotated (..) method. The name indicates that the method would check if the given solution were rotated correctly. But the method does something else. Name the methods appropriately. Coming to the Rotate (..) method, is it the array that is being rotated or the cells in the solution. What is the angle by which they are rotated? Is it 90 or 180 degrees? Please add proper description and name the methods appropriately.